openmc/lattice.py

```diff
@@ -0,0 +1,1901 @@
+                        uarray[-1].append([])
+
+                        # Climb down the bottom-right.
+                        for i in range(r):
+                            uarray[-1][-1].append(universe_ids[z, y, a])
+                            y -= 1
+
+                        # Climb across the bottom.
+                        for i in range(r):
+                            uarray[-1][-1].append(universe_ids[z, y, a])
+                            a -= 1
+
+                        # Climb up the bottom-left.
+                        for i in range(r):
+                            uarray[-1][-1].append(universe_ids[z, y, a])
+                            a -= 1
+                            y +=1
+
+                        # Climb up the top-left.
+                        for i in range(r):
+                            uarray[-1][-1].append(universe_ids[z, y, a])
+                            y += 1
+
+                        # Climb across the top.
+                        for i in range(r):
+                            uarray[-1][-1].append(universe_ids[z, y, a])
+                            a += 1
+
+                        # Climb down the top-right.
+                        for i in range(r):
+                            uarray[-1][-1].append(universe_ids[z, y, a])
+                            a += 1
+                            y -= 1
+
+                        # Move down to the next ring.
+                        a -= 1
+
+                        # Convert the ids into Universe objects.
+                        uarray[-1][-1] = [universes[u_id]
+                                          for u_id in uarray[-1][-1]]
+
+                    # Handle the degenerate center ring separately.
+                    u_id = universe_ids[z, y, a]
+                    uarray[-1].append([universes[u_id]])
+
+            # Add the universes to the lattice.
+            if len(pitch) == 2:
+                # Lattice is 3D
+                lattice.universes = uarray
+            else:
+                # Lattice is 2D; extract the only axial level
+                lattice.universes = uarray[0]
+
+        return lattice
+
+    def get_unique_universes(self):
+        """Determine all unique universes in the lattice
+
+        Returns
+        -------
+        universes : collections.OrderedDict
+            Dictionary whose keys are universe IDs and values are
+            :class:`openmc.Universe` instances
+
+        """
+
+        univs = OrderedDict()
+        for k in range(len(self._universes)):
+            for j in range(len(self._universes[k])):
+                if isinstance(self._universes[k][j], openmc.Universe):
+                    u = self._universes[k][j]
+                    univs[u._id] = u
+                else:
+                    for i in range(len(self._universes[k][j])):
+                        u = self._universes[k][j][i]
+                        assert isinstance(u, openmc.Universe)
```

Comment:
If this file is going to be compiled to bytecode (using `python -o` producing a `*.pyo` file), the assertion statement will be optimized out. 

Remediation:
Replace assertion with,

```
if not isinstance(u, openmc.Universe):
   raise AssertionError(...)
```